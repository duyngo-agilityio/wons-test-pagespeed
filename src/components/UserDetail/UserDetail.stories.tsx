import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

// Components
import UserDetail from './index';

const meta: Meta<typeof UserDetail> = {
  title: 'Components/UserDetail',
  component: UserDetail,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  decorators: [
    (Story) => (
      <div className="w-[600px]">
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof UserDetail>;

export const Default: Story = {
  args: {
    avatar:
      'https://www.hollywoodreporter.com/wp-content/uploads/2024/02/Avatar__The_Last_Airbender_n_S1_00_13_15_10RC.jpgAvatar__The_Last_Airbender_n_S1_00_13_15_10RC-H-2024.jpg?w=1296',
    username: 'airbender',
    role: 'user',
    fullName: 'Aang',
    email: 'aang@gmail.com',
    onButtonEditClick: fn(),
  },
};
