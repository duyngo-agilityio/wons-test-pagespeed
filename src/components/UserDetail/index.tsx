'use client';

import { memo } from 'react';

// Components
import { Text, Button, ImageFallback } from '@/components';

// Types
interface UserDetailProps {
  avatar: string;
  username: string;
  role: string;
  fullName: string;
  email: string;
  onButtonEditClick: () => void;
}

const UserDetail = ({
  avatar,
  username,
  role,
  fullName,
  email,
  onButtonEditClick,
}: UserDetailProps) => (
  <div className="p-[0_30px_0] w-full gap-[5px_0]">
    <div className="m-[30px_0] flex flex-col sm:flex-row justify-between items-center">
      <div className="sm:flex gap-[0_20px]">
        <div className="relative w-[128px] h-[128px]">
          <ImageFallback
            className="rounded-full object-cover h-full"
            fill
            alt="User Avatar"
            src={avatar}
          />
        </div>

        <dl className="flex flex-col justify-center text-center sm:text-left m-[20px_0] sm:m-0">
          <Text
            className="font-bold capitalize"
            size="2xl"
            as="dt"
            text={fullName}
          />
          <Text className="text-gray-900/80" size="md" as="dd" text={role} />
        </dl>
      </div>

      <Button
        className="h-fit text-[15px] font-medium md:w-auto py-[10px] px-[25px]"
        type="button"
        color="primary"
        onClick={onButtonEditClick}
      >
        Edit Profile
      </Button>
    </div>

    <dl className="flex flex-col gap-[2px_0] p-[15px_0] border-solid border-b-1 border-gray-100">
      <Text className="font-medium" size="xl" as="dt" text="User Name" />
      <Text className="text-gray-900/80" size="md" as="dd" text={username} />
    </dl>

    <dl className="flex flex-col gap-[2px_0] p-[15px_0] border-solid border-b-1 border-gray-100">
      <Text className="font-medium" size="xl" as="dt" text="Email" />
      <Text className="text-gray-900/80" size="md" as="dd" text={email} />
    </dl>
  </div>
);

export default memo(UserDetail);
