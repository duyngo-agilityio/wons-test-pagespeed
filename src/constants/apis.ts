export const API_PATH = {
  INVOICE_PRODUCTS: '/invoice-products',
  CALENDAR_TASKS: '/calendar-tasks',
  LOCATION_SUGGESTION: '/api/location-suggestion',
  PRODUCTS: '/products',
  SIGN_IN: '/auth/local',
  SIGN_UP: '/auth/local/register',
  STATISTICS: '/statistics',
  USERS: '/users',
  INVOICES: '/invoices',
  INVOICE: '/invoice',
  CUSTOMERS: '/customers',
  CUSTOMER: '/customer',
  TASKS: '/tasks',
  EVENTS: '/events',
};

export const STATUS_CODE = {
  OK: 200,
  CREATED: 201,
  ACCEPTED: 202,
  NO_CONTENT: 204,
  INVALID_PARAM: 400,
  NOT_FOUND: 404,
  CONFLICT: 409,
  INTERNAL_SERVER_ERROR: 500,
};
