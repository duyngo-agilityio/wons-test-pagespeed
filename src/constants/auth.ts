// Types
import { TRole } from '@/types';

export const AUTH_METHODS = {
  CREDENTIALS: 'credentials',
};

export const ERROR_TYPES = {
  CREDENTIALS_SIGN_IN: 'CallbackRouteError',
};

export const ROLES: TRole[] = [
  {
    id: 3,
    name: 'Admin',
  },
  {
    id: 4,
    name: 'User',
  },
];
