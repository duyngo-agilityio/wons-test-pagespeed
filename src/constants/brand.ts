export const BRANDS = [
  { key: 'apple', label: 'Apple' },
  { key: 'samsung', label: 'Samsung' },
  { key: 'huawei', label: 'Huawei' },
  { key: 'xioami', label: 'Xioami' },
  { key: 'oppo', label: 'Oppo' },
  { key: 'google', label: 'Google' },
];
