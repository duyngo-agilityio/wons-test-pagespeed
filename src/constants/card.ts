export enum LevelCard {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}
