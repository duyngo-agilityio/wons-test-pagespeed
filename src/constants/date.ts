export const DEFAULT_RANGE_VALUE_PICKER = 7;
