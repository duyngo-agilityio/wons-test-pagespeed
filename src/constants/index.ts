export * from './apis';
export * from './auth';
export * from './date';
export * from './fonts';
export * from './images';
export * from './messages';
export * from './params';
export * from './regex';
export * from './routes';
export * from './themes';
export * from './typography';
export * from './messages';
export * from './regex';
export * from './apis';
export * from './auth';
export * from './invoices';
export * from './select';
export * from './pagination';
export * from './brand';
export * from './rating';
export * from './tabs';
export * from './level';
export * from './taskStatus';
export * from './limits';
export * from './env';
