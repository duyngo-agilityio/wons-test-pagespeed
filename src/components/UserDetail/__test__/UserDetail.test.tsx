import { render } from '@testing-library/react';

// Constants
import { IMAGES } from '@/constants';

// Components
import UserDetail from '../index';

const mockUser = {
  avatar: IMAGES.AVATAR_DEFAULT,
  username: 'admin',
  fullName: 'Super Admin',
  email: 'admin1@gmail.com',
  role: 'Admin',
};

const onButtonEditClickMock = jest.fn();

describe('UserDetail', () => {
  const renderUserDetailComponent = () =>
    render(
      <UserDetail
        avatar={mockUser.avatar}
        username={mockUser.username}
        role={mockUser.role}
        email={mockUser.email}
        fullName={mockUser.fullName}
        onButtonEditClick={onButtonEditClickMock}
      />,
    );

  it('render successful', () => {
    const { container } = renderUserDetailComponent();

    // Check if the container is in the document
    expect(container).toBeInTheDocument();

    // Match the snapshot
    expect(container).toMatchSnapshot();
  });
});
