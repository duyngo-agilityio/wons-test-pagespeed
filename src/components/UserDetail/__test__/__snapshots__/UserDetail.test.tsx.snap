// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail render successful 1`] = `
<div>
  <div
    class="p-[0_30px_0] w-full gap-[5px_0]"
  >
    <div
      class="m-[30px_0] flex flex-col sm:flex-row justify-between items-center"
    >
      <div
        class="sm:flex gap-[0_20px]"
      >
        <div
          class="relative w-[128px] h-[128px]"
        >
          <img
            alt="User Avatar"
            class="rounded-full object-cover h-full"
            data-nimg="fill"
            decoding="async"
            fetchpriority="high"
            sizes="100vw"
            src="http://localhost/_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
        </div>
        <dl
          class="flex flex-col justify-center text-center sm:text-left m-[20px_0] sm:m-0"
        >
          <dt
            class="font-dm-sans text-blue-800 dark:text-white text-2xl font-bold capitalize"
          >
            Super Admin
          </dt>
          <dd
            class="font-dm-sans text-blue-800 dark:text-white text-md text-gray-900/80"
          >
            Admin
          </dd>
        </dl>
      </div>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 rounded-[10px] border-[1px] min-w-max h-fit text-[15px] font-medium md:w-auto py-[10px] px-[25px]"
        type="button"
      >
        <div
          class="flex items-center"
        >
          Edit Profile
        </div>
      </button>
    </div>
    <dl
      class="flex flex-col gap-[2px_0] p-[15px_0] border-solid border-b-1 border-gray-100"
    >
      <dt
        class="font-dm-sans text-blue-800 dark:text-white text-xl font-medium"
      >
        User Name
      </dt>
      <dd
        class="font-dm-sans text-blue-800 dark:text-white text-md text-gray-900/80"
      >
        admin
      </dd>
    </dl>
    <dl
      class="flex flex-col gap-[2px_0] p-[15px_0] border-solid border-b-1 border-gray-100"
    >
      <dt
        class="font-dm-sans text-blue-800 dark:text-white text-xl font-medium"
      >
        Email
      </dt>
      <dd
        class="font-dm-sans text-blue-800 dark:text-white text-md text-gray-900/80"
      >
        admin1@gmail.com
      </dd>
    </dl>
  </div>
</div>
`;
